src/main/capi/stream-c.cpp

```diff
@@ -7,19 +7,28 @@ duckdb_data_chunk duckdb_stream_fetch_chunk(duckdb_result result) {
 		return nullptr;
 	}
 	auto &result_data = *((duckdb::DuckDBResultData *)result.internal_data);
-	if (result_data.result_set_type == duckdb::CAPIResultSetType::CAPI_RESULT_TYPE_DEPRECATED) {
-		return nullptr;
-	}
 	if (result_data.result->type != duckdb::QueryResultType::STREAM_RESULT) {
 		// We can only fetch from a StreamQueryResult
 		return nullptr;
 	}
-	result_data.result_set_type = duckdb::CAPIResultSetType::CAPI_RESULT_TYPE_STREAMING;
-	auto &streaming = (duckdb::StreamQueryResult &)*result_data.result;
```

Comment:
Are we sure we are not losing functionality by removing the specific interrogation methods?

I am all for removing the ties between duckdb_result and MaterializedQueryResult

But does it not make sense to keep the specialized methods?
We already have a polymorphic query result class in C++, the translation to specialized methods in the C layer is minimal

Reply:
Or is this taking a python approach of "ask forgiveness, not permission"
Letting it throw if if wasnt open

Reply:
this code is still there, but now lives in `duckdb_fetch_chunk` which is called by the "old" `duckdb_stream_fetch_chunk" ?